backend/cfg/cfg_liveness.ml

```diff
@@ -18,71 +32,75 @@ module Domain = struct
+        let before = Reg.add_set_array across instr.arg in
+        { before; across }
     | Reloadretaddr ->
-      ( Reg.diff_set_array value Proc.destroyed_at_reloadretaddr,
-        Cfg.set_live instr Reg.Set.empty )
-    | Pushtrap _ -> value, Cfg.set_live instr Reg.Set.empty
-    | Poptrap -> value, Cfg.set_live instr Reg.Set.empty
-    | Prologue -> value, Cfg.set_live instr Reg.Set.empty
+      { before = Reg.diff_set_array before Proc.destroyed_at_reloadretaddr;
+        across = Reg.Set.empty
+      }
+    | Pushtrap _ -> { before; across = Reg.Set.empty }
+    | Poptrap -> { before; across = Reg.Set.empty }
+    | Prologue -> { before; across = Reg.Set.empty }
 
   let terminator :
-      domain ->
-      exn:domain ->
-      Cfg.terminator Cfg.instruction ->
-      domain * Cfg.terminator Cfg.instruction =
-   fun value ~exn instr ->
+      domain -> exn:domain -> Cfg.terminator Cfg.instruction -> domain =
+   fun { before; across = _ } ~exn instr ->
     match instr.desc with
     | Never -> assert false
-    | Always _ -> Reg.add_set_array value instr.arg, Cfg.set_live instr value
+    | Always _ ->
+      { before = Reg.add_set_array before instr.arg; across = before }
     | Parity_test _ ->
-      Reg.add_set_array value instr.arg, Cfg.set_live instr value
+      { before = Reg.add_set_array before instr.arg; across = before }
     | Truth_test _ ->
-      Reg.add_set_array value instr.arg, Cfg.set_live instr value
+      { before = Reg.add_set_array before instr.arg; across = before }
     | Float_test _ ->
-      Reg.add_set_array value instr.arg, Cfg.set_live instr value
-    | Int_test _ -> Reg.add_set_array value instr.arg, Cfg.set_live instr value
-    | Switch _ -> Reg.add_set_array value instr.arg, Cfg.set_live instr value
-    | Return -> Reg.set_of_array instr.arg, Cfg.set_live instr Reg.Set.empty
+      { before = Reg.add_set_array before instr.arg; across = before }
+    | Int_test _ ->
+      { before = Reg.add_set_array before instr.arg; across = before }
+    | Switch _ ->
+      { before = Reg.add_set_array before instr.arg; across = before }
+    | Return -> { before = Reg.set_of_array instr.arg; across = Reg.Set.empty }
     | Tailcall (Self _) ->
-      Reg.set_of_array instr.arg, Cfg.set_live instr Reg.Set.empty
-    | Raise _ -> Reg.add_set_array exn instr.arg, Cfg.set_live instr exn
+      { before = Reg.set_of_array instr.arg; across = Reg.Set.empty }
+    | Raise _ ->
+      { before = Reg.add_set_array exn.before instr.arg; across = exn.before }
     | Tailcall (Func _) ->
-      Reg.set_of_array instr.arg, Cfg.set_live instr Reg.Set.empty
+      { before = Reg.set_of_array instr.arg; across = Reg.Set.empty }
     | Call_no_return _ ->
-      Reg.add_set_array exn instr.arg, Cfg.set_live instr exn
+      { before = Reg.add_set_array exn.before instr.arg; across = exn.before }
 
   let exception_ : domain -> domain =
-   fun value -> Reg.Set.remove Proc.loc_exn_bucket value
+   fun { before; across } ->
```

Comment:
Can across in the input be ignore here (and output set to empty)?

Reply:
It should not change anything, since such `across`
values are over an exceptional edges that hold no
instruction.